### ISSUE TITLE
Symmetry unittest

### ISSUE THREAD

**Issue by JiazhengSun** [dartsim/dart#1193]
This is the unittest I did to check symmetry. 

```cpp
class SymTest : public testing::Test
{
  public:
    void AngularVelAdd();
  protected:
};

void SymTest::AngularVelAdd()
{
  WorldPtr world = World::create();
  EXPECT_TRUE(world != nullptr);
  world->setGravity(Vector3d(0.0, -10.0, 0.0));
  world->setTimeStep(0.001);

  SkeletonPtr sphereSkel = createSphere(0.05, Vector3d(0.0, 1.0, 0.0));
  BodyNode* sphere = sphereSkel->getBodyNode(0);
  Joint* sphereJoint = sphere->getParentJoint();
  sphereJoint->setVelocity(0, 10.0);  // ang_x -> Affect lz and ly
  sphereJoint->setVelocity(1, 10.0);  // ang_y -> No effect
  sphereJoint->setVelocity(2, 10.0);  // ang_z -> Affect lx and ly
  world->addSkeleton(sphereSkel);

  Eigen::Vector3d LinearVelBefore = sphere->getLinearVelocity(); //Get linear velocity 

  cout<<"Linear Velocity before running is "<< LinearVelBefore.transpose()<<endl;

  int maxSteps = 500;
  for (int i = 0; i < maxSteps; i++)
  {
    //cout<<"linear Z is: "<<sphere->getLinearVelocity()[2]<<endl;
    world->step();
  }

  Vector3d LinearVelAfter = sphere->getLinearVelocity(); // Get linear velocity
  cout<<"Linear Velocity after running is "<<LinearVelAfter.transpose()<<endl;
  double lx = LinearVelAfter[0];
  //double ly = LinearVelAfter[1];
  double lz = LinearVelAfter[2];

  EXPECT_EQ(lx, 0.0);
  //EXPECT_EQ(ly, 0.0);
  EXPECT_EQ(lz, 0.0);
}

TEST_F(SymTest, AngularVelAdd)
{
  AngularVelAdd();
}

```

**Comment by jslee02** [comment]
Thank you for sharing the test code. Let me investigate this.

**Comment by azeey** [comment]
I believe PR #1437 fixes this. I've added a regression test based on @JiazhengSun's code.

**Comment by jslee02** [comment]
Finally resolved by #1437
